### PR TITLE
use Path.Combine for cross-platform support

### PR DESCRIPTION
**Pull request by josch**
Hi,

using mono, RSCacheTool runs on Linux after applying this patch. Please consider applying it.

Thanks!

### CODE REVIEW

**Comment by josch** [comment]
Ah one note I forgot to mention. I had to rename "sox.exe" to "sox" because the ".exe" prefix is only used in windows but not on other operating systems. It should probably made such that the sox binary name is chosen depending on the platform the program is running on. But since is the first time I wrote c# I'm not sure how to do this properly.

**Comment by Villermen** [comment]
Eya josch,

I have changed all backslashes to forward-slashes, which would (probably, untested) make it compatible with other platforms while still working on windows. Please tell me if this actually does. Also, about the sox.exe: Have you installed sox as a package and will it run that program with the sox command? If so it should probably be easy to change, but harder to inform the user about. I try make this package as complete as possible (hence including the sox.exe in the download). If there's a separate binary file of sox for linux (/mac) I'll try to include that and make the program launch whichever is needed.

So to summarize I have tried to fix the path separator thing by changing them to forward-slashes and need to know how linux handles running of sox (is it installed as a package or is there some other magic going on?).

Thank you for helping to improve the program!

Gr.Viller

**Comment by josch** [comment]
Why not use Path.Combine?

I have installed sox as a package and it will run it when called as "sox". I cannot say how it is named on mac.

**Comment by Villermen** [comment]
I have bad experiences with Path.Combine (e.g. using an absolute path as a non-first parameter will make the output start at that path, which is not desired, especially when considering user input). You can read up about this [here](http://thingsihateaboutmicrosoft.blogspot.nl/2009/08/pathcombine-is-essentially-useless.html?m=1).

I'll add sox to the install command in the 'Compiling under linux' section just to make sure it is present for the user building the program. A binary for mac exists and I'll just throw it in there too.

**Comment by Villermen** [comment]
This should make it work, can you confirm?

**Comment by josch** [comment]
I confirm that these changes make it work on Debian GNU/Linux

**Comment by Villermen** [comment]
Awesome, thanks again. I'll see to your other request later today.
